### DISCUSSION TITLE
Modbus 485 connection with the USR-W610 does not work

### DISCUSSION THREAD

**Discussion by BrewDok** [StealthChesnut/HA-FoxESS-Modbus#125] [General]
Hi everyone,

i am an owner of the Fox H3-12.0 and very annoyed by the cloud availability and the api delay of 5 minutes. I have a proper connection of the system in home assistant via the FoxESS-Cloud and the lan dongle but the current downtime made me to overthink this setup. Then i found a fork of this project and connection of the H3 for HomeAssistant and instantly ordered the USR-W610 and followed the USR W610 Setup Guide. I came across an existing discussion here where the problem was a faulty wiring, but i tried numerous configurations with the same result and would rule this out. If i use modpoll to validate the connection i get a timeout with the following:
```
D:\modpoll\win>modpoll.exe -m tcp -p 502 -a 247 -t 3 -0 -1 -c 16 -r 11000 192.168.5.130
modpoll 3.10 - FieldTalk(tm) Modbus(R) Master Simulator
Copyright (c) 2002-2021 proconX Pty Ltd
Visit https://www.modbusdriver.com for Modbus libraries and tools.

Protocol configuration: MODBUS/TCP, FC4
Slave configuration...: address = 247, start reference = 11000 (PDU), count = 16
Communication.........: 192.168.5.130, port 502, t/o 1.00 s, poll rate 1000 ms
Data type.............: 16-bit register, input register table

-- Polling slave...
Reply time-out!
```
Since this is not my expertise and the setup guide does not really state what to look for,  i don´t really know where to troubleshoot and came up with the following questions:
- Is the "Reply time-out" ok`?
- Is there any way to validate pin 1 and 2 are delivering a modbus signal?
- Do i need to activate the datalogger Modbus485 functionality over Pin 1/2 in the cloud software or the inverter?
- Is a certain software version of the Inverter / Logger required?

Your help is very appreciated
André

**Comment by canton7** [comment]
> Is the "Reply time-out" ok`?

No

> Is there any way to validate pin 1 and 2 are delivering a modbus signal?

If you've got a scope you could check that the W610 is transmitting correctly, but the problem is that the inverter isn't responding

> Do i need to activate the datalogger Modbus485 functionality over Pin 1/2 in the cloud software or the inverter?

No

> Is a certain software version of the Inverter / Logger required?

No

Double-check that the settings on the W610 are correct. Check that you've connected the correct pins on the inverter connector to the correct terminals on the W610 (A and B on the RS485 port, **not** the RS232 port).

**Comment by Daibutt** [comment]
There is a comment here where default slave address was 000 not 247 for an H3....
https://www.facebook.com/groups/foxessownersgroup/permalink/1843062412742599/
